File: packages/lector/test/fixtures/typescript-reference/packages/app/src/checkout.ts
import type { Order, PaymentProcessor, Receipt } from "@fixture/contracts";

export async function runCheckout(processor: PaymentProcessor, order: Order): Promise<Receipt> {
	return processor.process(order);
}

export async function runCheckoutTwice(processor: PaymentProcessor, order: Order): Promise<readonly Receipt[]> {
	return Promise.all([runCheckout(processor, order), runCheckout(processor, order)]);
}

File: packages/lector/test/fixtures/rust-reference/app/src/checkout.rs
use contracts::{Order, PaymentProcessor, Receipt};

pub fn run_checkout(processor: &dyn PaymentProcessor, order: Order) -> Receipt {
    processor.process(order)
}

pub fn run_checkout_twice(processor: &dyn PaymentProcessor, order: Order) -> Receipt {
    run_checkout(processor, order.clone());
    run_checkout(processor, order)
}
